stratum/hal/lib/tdi/es2k/es2k_sde_meter.cc

```diff
@@ -0,0 +1,248 @@
+      } else if (field_name == kEs2kMeterCommitedBurstPackets) {
+        uint64 cburst;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &cburst));
+        cfg[i].cburst = KbitsToBytesPerSecond(cburst);
+      } else if (field_name == kEs2kMeterPirPps) {
+        uint64 pir;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &pir));
+        cfg[i].pir = KbitsToBytesPerSecond(pir);
+      } else if (field_name == kEs2kMeterPeakBurstPackets) {
+        uint64 pburst;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &pburst));
+        cfg[i].pburst = KbitsToBytesPerSecond(pburst);
+      } else if (field_name == kEs2kMeterCirKPpsUnit) {
+        uint64 cir_unit;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &cir_unit));
+        cfg[i].cir_unit = KbitsToBytesPerSecond(cir_unit);
+      } else if (field_name == kEs2kMeterCommitedBurstPacketsUnit) {
+        uint64 cburst_unit;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &cburst_unit));
+        cfg[i].cburst_unit = KbitsToBytesPerSecond(cburst_unit);
+      } else if (field_name == kEs2kMeterPirPpsUnit) {
+        uint64 pir_unit;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &pir_unit));
+        cfg[i].pir_unit = KbitsToBytesPerSecond(pir_unit);
+      } else if (field_name == kEs2kMeterPeakBurstPacketsUnit) {
+        uint64 pburst_unit;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &pburst_unit));
+        cfg[i].pburst_unit = KbitsToBytesPerSecond(pburst_unit);
+      } else if (field_name == kEs2kMeterGreenCounterBytes) {
+        uint64 greenBytes;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &greenBytes));
+        cfg[i].greenBytes = KbitsToBytesPerSecond(greenBytes);
+      } else if (field_name == kEs2kMeterGreenCounterPackets) {
+        uint64 greenPackets;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &greenPackets));
+        cfg[i].greenPackets = KbitsToBytesPerSecond(greenPackets);
+      } else if (field_name == kEs2kMeterYellowCounterBytes) {
+        uint64 yellowBytes;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &yellowBytes));
+        cfg[i].yellowBytes = KbitsToBytesPerSecond(yellowBytes);
+      } else if (field_name == kEs2kMeterYellowCounterPackets) {
+        uint64 yellowPackets;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &yellowPackets));
+        cfg[i].yellowPackets = KbitsToBytesPerSecond(yellowPackets);
+      } else if (field_name == kEs2kMeterRedCounterBytes) {
+        uint64 redBytes;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &redBytes));
+        cfg[i].redBytes = KbitsToBytesPerSecond(redBytes);
+      } else if (field_name == kEs2kMeterRedCounterPackets) {
+        uint64 redPackets;
+        RETURN_IF_TDI_ERROR(table_data->getValue(field_id, &redPackets));
+        cfg[i].redPackets = KbitsToBytesPerSecond(redPackets);
+      } else {
+        MAKE_ERROR(ERR_INVALID_PARAM)
```

Comment:
```suggestion
        return MAKE_ERROR(ERR_INVALID_PARAM)
```
Otherwise, the statement is useless.

Reply:
Fixed